docs/src/main/tut/dsl.md

```diff
@@ -214,25 +214,26 @@ import scala.concurrent.Future
 import scala.concurrent.ExecutionContext.Implicits.global
 ```
 
-You can seamlessly respond with a `Future` of any type that has an
-`EntityEncoder`.
+You can respond with a `Future` of any type that has an
+`EntityEncoder` by lifting it into IO. Note: `Future` is a side effecting and
+impure member of the standard library, thus we must defer its evaluation
 
 ```tut
-val io = Ok(Future {
+val io = IO.fromFuture(IO(Future {
   println("I run when the future is constructed.")
   "Greetings from the future!"
-})
+})).flatMap(Ok(_))
 io.unsafeRunSync
 ```
 
 As good functional programmers who like to delay our side effects, we
 of course prefer to operate in [F]s:
 
 ```tut
-val io = Ok(IO {
+val io = IO {
```

Comment:
Would it be feasible to allow the old syntax with deprecation in http4s DSL?  It would require something like:

```
@deprecated("flatMap that shit", "0.19")
def apply[F[_]](a: F[A])(implicit EE: EntityEncoder[F, A])
```

This overload is probably not going to fly with the existing one, but it's worth a shot if we can do compiler-assisted migration.  This is a big breaking change.

Alternatively, it would be neat if we could target and scalafix the ones we ate.